MeetupPlatformApi/Controllers/AuthenticationController.cs

```diff
@@ -0,0 +1,74 @@
+﻿namespace MeetupPlatformApi.Controllers;
+
+using AutoMapper;
+using MeetupPlatformApi.Authentication;
+using MeetupPlatformApi.Context;
+using MeetupPlatformApi.DataTransferObjects;
+using MeetupPlatformApi.Entities;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using System.Security.Claims;
+using BCrypt.Net;
+
+[Route("/api/users")]
+[ApiController]
+public class AuthenticationController : ControllerBase
+{
+    private readonly ApplicationContext context;
+    private readonly IMapper mapper;
+    private readonly AuthenticationManager authenticationManager;
+
+    public AuthenticationController(ApplicationContext context, IMapper mapper, AuthenticationManager authenticationManager)
+    {
+        this.context = context;
+        this.mapper = mapper;
+        this.authenticationManager = authenticationManager;
+    }
+
+    [HttpGet("{id:guid}")]
+    public async Task<IActionResult> GetUserById(Guid id)
+    {
+        var user = await context.Users.SingleOrDefaultAsync(user => user.Id == id);
+        var outputDto = mapper.Map<UserOutputDto>(user);
+        return outputDto is null ? Ok(outputDto) : NotFound();
+    }
+
+    [HttpPost]
+    public async Task<IActionResult> RegisterUser([FromBody] UserRegistrationDto userFromBody)
+    {
+        var user = mapper.Map<UserEntity>(userFromBody);
+        user.Password = BCrypt.HashPassword(user.Password);
+
+        context.Users.Add(user);
+        await context.SaveChangesAsync();
+
+        var token = authenticationManager.CreateToken(user);
+        var outputDto = mapper.Map<UserOutputDto>(user);
```

Comment:
In this case `UserRegistrationResultDto` is our main output DTO, so I recommend refactoring code a bit:
```csharp
var userInfoDto = mapper.Map<UserOutputDto>(user);
var outputDto = new UserRegistrationResultDto { UserInfo = userInfoDto , AccessToken = token };
return CreatedAtAction(nameof(GetUserById), new { id = outputDto.UserInfo.Id }, outputDto);
```